src/neuroconv/tools/roiextractors/roiextractors.py

```diff
@@ -468,66 +608,29 @@ def write_segmentation(
-        # PlaneSegmentation:
-        input_kwargs = dict(
-            description="output from segmenting imaging plane",
-            imaging_plane=imaging_plane,
-        )
-        ps_metadata = metadata["Ophys"]["ImageSegmentation"]["plane_segmentations"][0]
-        if ps_metadata["name"] not in image_segmentation.plane_segmentations:
-            ps_exist = False
-        else:
-            ps = image_segmentation.get_plane_segmentation(ps_metadata["name"])
-            ps_exist = True
-        roi_ids = segext_obj.get_roi_ids()
-        accepted_list = segext_obj.get_accepted_list()
-        accepted_list = [] if accepted_list is None else accepted_list
-        rejected_list = segext_obj.get_rejected_list()
-        rejected_list = [] if rejected_list is None else rejected_list
-        accepted_ids = [1 if k in accepted_list else 0 for k in roi_ids]
-        rejected_ids = [1 if k in rejected_list else 0 for k in roi_ids]
-        roi_locations = np.array(segext_obj.get_roi_locations()).T
-
-        def image_mask_iterator():
-            for id in segext_obj.get_roi_ids():
-                img_msks = segext_obj.get_roi_image_masks(roi_ids=[id]).T.squeeze()
-                yield img_msks
-
-        if not ps_exist:
-            from hdmf.common import VectorData
-
-            input_kwargs.update(
-                **ps_metadata,
-                columns=[
-                    VectorData(
-                        data=H5DataIO(
-                            DataChunkIterator(image_mask_iterator(), buffer_size=buffer_size),
-                            compression=True,
-                            compression_opts=9,
-                        ),
-                        name="image_mask",
-                        description="image masks",
-                    ),
-                    VectorData(
-                        data=roi_locations,
-                        name="RoiCentroid",
-                        description="x,y location of centroid of the roi in image_mask",
-                    ),
-                    VectorData(
-                        data=accepted_ids,
-                        name="Accepted",
-                        description="1 if ROi was accepted or 0 if rejected as a cell during segmentation operation",
-                    ),
-                    VectorData(
-                        data=rejected_ids,
-                        name="Rejected",
-                        description="1 if ROi was rejected or 0 if accepted as a cell during segmentation operation",
-                    ),
-                ],
-                id=roi_ids,
+
+        # Add plane segmentation
+        plane_segmentation_metadata = get_plane_segmentation_metadata(metadata=metadata)
+        plane_segmentation_name = plane_segmentation_metadata["name"]
+        if plane_segmentation_name not in image_segmentation.plane_segmentations:
+            segmentation_results = _get_output_from_segmentation_extractor(
```

Comment:
I feel that everything from here to lines 627 should happen inside `add_plane_segmentation`. The equivalent function in the spikeinterface module  is `add_electrodes`. 

There, the information necessary to create the dynamic table is extracted from the sorting_extractor (which is an argument of the function) which here is done in your `get_output_from_segmentation_extractor` function. The `add_electrodes` function also checks for duplicates and performs the addition of the objects which here is done in your `add_plane_segmentation` function. 

I think that the functions to add the corresponding dynamic table in the two pipe lines should behave in a similar way and this is where the suggestion is coming from.

How are you thinking about it?